test/test_helper.rb

```diff
@@ -30,6 +30,15 @@ def with_preview_route(new_value)
   app.reloader.reload!
 end
 
+def with_preview_base_controller(new_value)
+  old_value = Rails.application.config.view_component.preview_base_controller
+  Rails.application.config.view_component.preview_base_controller = new_value
+  app.reloader.reload!
+  yield
+  Rails.application.config.view_component.preview_base_controller = old_value
+  app.reloader.reload!
+end
+
```

Comment:
hmmm ok so it seems that the reloader doesn't actually reloads the previews controller hierarchy.

I'm fine with leaving this feature without tests for the time being then, until we come up with a way to cover it.

One shady suggestion might be to remove the definition of the controller `Object.send(:remove_const, 'ViewComponentsController')` and then require the file manually by doing `require 'app/controllers/view_components_controller`. I'm not sure that would work, but I'm sure it would look ugly :see_no_evil: